packages/relay/src/lib/services/ethService/ethFilterService/index.ts

```diff
@@ -167,11 +172,56 @@ export class FilterService implements IFilterService {
 
     return this.common.getLogs(
       null,
-      filter?.params.fromBlock,
+      useLastQueried ? filter?.lastQueried : filter?.params.fromBlock,
       filter?.params.toBlock,
       filter?.params.address,
       filter?.params.topics,
       requestIdPrefix
     );
   }
+
+  public async getFilterChanges(filterId: string, requestIdPrefix?: string): Promise<string[] | Log[] | JsonRpcError> {
+    this.logger.trace(`${requestIdPrefix} getFilterChanges(${filterId})`);
+    FilterService.requireFiltersEnabled();
+
+    const cacheKey = `${constants.CACHE_KEY.FILTERID}_${filterId}`;
+    const filter = this.cache.get(cacheKey, this.ethGetFilterChanges, requestIdPrefix);
+
+    if (!filter) {
+      throw predefined.FILTER_NOT_FOUND;
+    }
+
+    let result, latestBlockNumber;
+
+    if (filter.type === constants.FILTER.TYPE.LOG) {
+      result = await this.getFilterLogs(filterId, true);
+
+      // get the latest block number and add 1 to exclude current results from the next response because
+      // the mirror node query executes "gte" not "gt"
+      latestBlockNumber = Number(
+        result.length ? result[result.length - 1].blockNumber : await this.common.getLatestBlockNumber(requestIdPrefix)
+      ) + 1;
+    } else if (filter.type === constants.FILTER.TYPE.NEW_BLOCK) {
+      result = await this.mirrorNodeClient.getBlocks([
+        `gte:${filter.lastQueried || filter.params.blockAtCreation}`
+      ], undefined, {
+        order: 'asc'
+      });
+
+      latestBlockNumber = Number(
+        result?.blocks?.length ? result.blocks[result.blocks.length - 1].number : await this.common.getLatestBlockNumber(requestIdPrefix)
+      );
+
+      result = result?.blocks?.map(r => r.hash) || [];
+    }
```

Comment:
add an else that handles if filter.type is none of the supported. Also add test case to confirm

Reply:
I don't see a UT case for this